### PR TITLE
Specify validation dataset

### PR DESCRIPTION
**Pull request by charlielutra**
When calling MACE.train(), rather than taking a single input ConfigurationSet to be the training data, and then taking a random 10% (by default - set as --valid_fraction) of that dataset as a validation set; now there's the option to have a second ConfigSet as input which will be the validation set of the potential. 

Example use case: if there are two or more distinct datasets being used to train the model: you can (beforehand) perform a stratified split of these, to ensure an even split of these datasets in the training vs validation datasets, so that the model can learn from both effectively.

### CODE REVIEW

**Comment by juraskov** [comment]
As agreed in the comment in #199, closing this pull request and will continue with #199 instead
